third_party/pcsc-lite/naclport/server_clients_management/src/clients_manager.cc

```diff
@@ -146,15 +169,35 @@ bool PcscLiteServerClientsManager::DeleteHandlerMessageListener ::
   return true;
 }
 
+PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    UpdateAdminPolicyMessageListener(AdminPolicyGetter* admin_policy_getter)
+    : admin_policy_getter_(admin_policy_getter) {}
+
+std::string PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    GetListenedMessageType() const {
+  return kUpdateAdminPolicyMessageType;
+}
+
+bool PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    OnTypedMessageReceived(Value data) {
+  const UpdateAdminPolicyMessageData message_data =
+      ConvertFromValueOrDie<UpdateAdminPolicyMessageData>(std::move(data));
```

Comment:
Here we shouldn't use the "OrDie" variant because it'll crash the program if someone sets a misconfigured policy. Let's use `ConvertFromValue()` and print a warning log when there's a conversion issue.